test/e2e/resource-filtering/exclude_label.go

```diff
@@ -0,0 +1,129 @@
+	rand.Seed(time.Now().UnixNano())
+	UUIDgen, _ = uuid.NewRandom()
+	basicCase := FilteringCase{}
+	basicCase = basicCase.Init(client, isTestBackup)
+	e = ExcludeFromBackup{
+		FilteringCase: basicCase,
+	}
+	e.backupName = "backup-exclude-from-backup-" + UUIDgen.String()
+	e.restoreName = "restore-exclude-from-backup-" + UUIDgen.String()
+	e.baseName = "exclude-from-backup-" + UUIDgen.String()
+	e.Desc = "Backup with the label velero.io/exclude-from-backup=true are not included test"
+	e.Text = "Should not backup resources with the label velero.io/exclude-from-backup=true"
+	e.FaildMsg = "Failed to backup resources with the label velero.io/exclude-from-backup=true"
+	e.Ctx = timeout
+	e.labels = map[string]string{
+		"velero.io/exclude-from-backup": "true",
+	}
+	e.labelSelector = "velero.io/exclude-from-backup"
+	e.backupArgs = []string{
+		"create", "--namespace", VeleroCfg.VeleroNamespace, "backup", e.backupName,
+		"--default-volumes-to-restic", "--wait",
+	}
+
+	e.restoreArgs = []string{
+		"create", "--namespace", VeleroCfg.VeleroNamespace, "restore", e.restoreName,
+		"--from-backup", e.backupName, "--wait",
+	}
+	return e
+}
+
+func (e ExcludeFromBackup) CreateResources() error {
+	for nsNum := 0; nsNum < e.namespacesTotal; nsNum++ {
+		namespace := fmt.Sprintf("%s-%00000d", e.baseName, nsNum)
+		fmt.Printf("Creating resources in namespace ...%s\n", namespace)
+		labels := e.labels
+		if nsNum%2 == 0 {
+			labels = map[string]string{
+				"velero.io/exclude-from-backup": "false",
+			}
+		}
+		if err := CreateNamespaceWithLabel(e.Ctx, e.Client, namespace, labels); err != nil {
+			return errors.Wrapf(err, "Failed to create namespace %s", namespace)
+		}
+
+		//Create deployment
+		fmt.Printf("Creating deployment in namespaces ...%s\n", namespace)
+
+		deployment := NewDeployment(e.baseName, namespace, e.replica, labels)
+		deployment, err := CreateDeployment(e.Client.ClientGo, namespace, deployment)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to delete the namespace %q", namespace))
+		}
+		err = WaitForReadyDeployment(e.Client.ClientGo, namespace, deployment.Name)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to ensure deployment completion in namespace: %q", namespace))
+		}
+	}
+	return nil
+}
+
+func (e ExcludeFromBackup) Verify() error {
+	for nsNum := 0; nsNum < e.namespacesTotal; nsNum++ {
+		namespace := fmt.Sprintf("%s-%00000d", e.baseName, nsNum)
+		fmt.Printf("Checking resources in namespaces ...%s\n", namespace)
+		//Check deployment
+		_, err := GetDeployment(e.Client.ClientGo, namespace, e.baseName)
+		if nsNum%2 == 0 { //include
+			if err != nil {
+				return errors.Wrap(err, fmt.Sprintf("failed to list deployment in namespace: %q", namespace))
+			}
+		} else { //exclude
+			if err == nil {
+				return fmt.Errorf("failed to exclude deployment in namespaces %q", namespace)
+			} else {
+				if strings.Contains(err.Error(), "not found") { //resource should be excluded
```

Comment:
Is it possible to use the standard error definitions to check the not found error?

Reply:
ok